legacy/src/dnn_backend/dnn_backend_miopen.cpp.inc

```diff
@@ -0,0 +1,1236 @@
+    ConvolutionDescriptor_t const& conv_desc,
+    ConvBwdDataAlgo_t const& conv_algo,
+    void* work_data,
+    size_t work_data_size,
+    void const* beta,
+    TensorDescriptor_t const& dx_desc,
+    void* dx_data)
+{
+    auto output_helper =
+        handle_alpha_beta(handle, alpha, beta, dx_desc, dx_data);
+    DISTCONV_CHECK_MIOPEN(miopenConvolutionBackwardData(handle,
+                                                        &one,
+                                                        dy_desc,
+                                                        dy_data,
+                                                        filter_desc,
+                                                        filter_data,
+                                                        conv_desc,
+                                                        conv_algo,
+                                                        &zero,
+                                                        dx_desc,
+                                                        output_helper,
+                                                        work_data,
+                                                        work_data_size));
+}
+
+void GPUDNNBackend::convolution_bwd_filter(
+    Handle_t handle,
+    void const* alpha,
+    TensorDescriptor_t const& in_desc,
+    void const* in_data,
+    TensorDescriptor_t const& dy_desc,
+    void const* dy_data,
+    ConvolutionDescriptor_t const& conv_desc,
+    ConvBwdFilterAlgo_t const& conv_algo,
+    void* work_data,
+    size_t work_data_size,
+    void const* beta,
+    FilterDescriptor_t const& dw_desc,
+    void* dw_data)
+{
+    auto output_helper =
+        handle_alpha_beta(handle, alpha, beta, dw_desc, dw_data);
+    DISTCONV_CHECK_MIOPEN(miopenConvolutionBackwardWeights(handle,
+                                                           &one,
+                                                           dy_desc,
+                                                           dy_data,
+                                                           in_desc,
+                                                           in_data,
+                                                           conv_desc,
+                                                           conv_algo,
+                                                           &zero,
+                                                           dw_desc,
+                                                           output_helper,
+                                                           work_data,
+                                                           work_data_size));
+}
+
+size_t GPUDNNBackend::get_conv_forward_workspace_size(
+    Handle_t const& /*handle*/,
+    TensorDescriptor_t const& /*in_desc*/,
+    FilterDescriptor_t const& /*filter_desc*/,
+    ConvolutionDescriptor_t const& /*conv_desc*/,
+    TensorDescriptor_t const& /*out_desc*/,
+    ConvFwdAlgo_t const& /*algo*/)
+{
+    return 1 << 30;
```

Comment:
:(

Reply:
This is here for a few reasons:

1. This is what we do in LBANN, and it's what we were doing prior to this PR (doesn't make it right, just consistent).
2. The MIOpen APIs for querying this from the library are VERY slow. We'd have to use some memoization anyway. This is just degenerate and pessimistic memoization.
3. There are subtle semantic inconsistencies between MIOpen/cuDNN that are harder to resolve than just returning a fixed value here.